src/caffeine/caffeine.c

```diff
@@ -287,7 +287,7 @@ size_t caf_elem_len(CFI_cdesc_t* a_desc)
   return a_desc->elem_len;
 }
 
-void caf_form_team(gex_TM_t current_team, gex_TM_t* new_team, intmax_t team_number, int new_index)
+void caf_form_team(gex_TM_t current_team, gex_TM_t* new_team, int64_t team_number, int new_index)
 {
   gex_TM_Split(new_team, current_team, team_number, new_index, NULL, 0, GEX_FLAG_TM_NO_SCRATCH);
```

Comment:
The third argument to `gex_TM_Split` has type `int`, so this call involves an implicit narrowing conversion of `team_number` from 64 to 32 bit, which will silently truncate anything in the upper bits.

Caffeine will not be handling this in the near future, so I'd like an assertion here, something of the form:

```suggestion
   // GASNet color argument is int (32-bit), check for value truncation:
  assert((unsigned int)team_number == team_number);
  gex_TM_Split(new_team, current_team, team_number, new_index, NULL, 0, GEX_FLAG_TM_NO_SCRATCH);
```

Reply:
This is not directly relevant to this PR, just an undocumented issue that's been on my radar for awhile.